### PR TITLE
Baro fix

### PR DESCRIPTION
**Pull request by tumbili**
- compute atmospheric pressure from altitude
- use correct report length to read baro data from simulator

### CODE REVIEW

**Comment by LorenzMeier** [comment]
I presume that's tested? Changes look good.

File: src/platforms/posix/drivers/barosim/baro_sim.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ BARO_SIM::dev_read(unsigned offset, void *data, unsigned count)
 
 	/* read the most recent measurement */
 	uint8_t cmd = 0;
-	int ret = transfer(&cmd, 1, static_cast<uint8_t *>(data), 3);
+	int ret = transfer(&cmd, 1, static_cast<uint8_t *>(data), count);
```
Comment (LorenzMeier) [inline src/platforms/posix/drivers/barosim/baro_sim.cpp]: Is count correct here or count * sizeof something?

**Comment by tumbili** [comment]
Yes, it's tested, ALTCLT works now in SITL

File: src/platforms/posix/drivers/barosim/baro_sim.cpp (hunk 1 above)
Comment (tumbili) [inline src/platforms/posix/drivers/barosim/baro_sim.cpp]: I was wondering myself, probably size_t is best?

Comment (tumbili) [inline src/platforms/posix/drivers/barosim/baro_sim.cpp]: Actually everywhere else "unsigned" is used, e.g. in gyrosim

**Pull request merged**